### PR TITLE
Miawong/sc 48390/kots edit config layout overlaps left menu

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it:
This change improves the responsiveness of the UI when user is on the app config page. 
Loom: https://www.loom.com/share/3d2b13c978054c1689e9239ab5787d97

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/48390/kots-edit-config-layout-overlaps-left-menu

#### Does this PR introduce a user-facing change?

```release-note
Improves the UI responsiveness on the configuration page. 

```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/Root.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import find from "lodash/find";
 import ConnectionTerminated from "./ConnectionTerminated";
 import GitOps from "././components/clusters/GitOps";
 import PreflightResultPage from "./components/PreflightResultPage";
-import AppConfig from "./components/apps/AppConfig";
+import AppConfig from "./features/AppConfig/components/AppConfig";
 import AppDetailPage from "./components/apps/AppDetailPage";
```
Comment (miaawong) [inline web/src/Root.jsx:11]: using the bulletproof folder structure

File: web/src/components/apps/Dashboard.jsx [hunk 2]
```diff
@@ -800,9 +799,9 @@ class Dashboard extends Component {
             className="Modal DefaultSize"
           >
             <div className="Modal-body">
-              <ParagraphLarge weight={"bold"} theme={theme}>
+              <Paragraph size="16" weight="bold">
```
Comment (miaawong) [inline web/src/components/apps/Dashboard.jsx:802]: decided to extract Paragraph styled component out for other usage.

File: web/src/features/AppConfig/components/AppConfig.jsx [hunk 3]
```diff
@@ -418,99 +428,6 @@ class AppConfig extends Component {
     this.setState({ showNextStepModal: false });
   };
 
-  renderConfigInfo = (app) => {
-    const { match, fromLicenseFlow } = this.props;
```
Comment (miaawong) [inline web/src/features/AppConfig/components/AppConfig.jsx:422]: I decided to extract this to it's own file see ConfigInfo.jsx

File: web/src/components/apps/Dashboard.jsx (hunk 2 above)
Comment (miaawong) [inline web/src/components/apps/Dashboard.jsx:802]: decided to extract Paragraph styled component out for other usage.

File: web/src/Root.jsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ import find from "lodash/find";
 import ConnectionTerminated from "./ConnectionTerminated";
 import GitOps from "././components/clusters/GitOps";
 import PreflightResultPage from "./components/PreflightResultPage";
-import AppConfig from "./components/apps/AppConfig";
+import AppConfig from "./features/AppConfig/components/AppConfig";
```
Comment (miaawong) [inline web/src/Root.jsx:10]: using the bulletproof folder structure, putting AppConfig into feature

File: web/src/features/AppConfig/components/AppConfig.jsx [hunk 5]
```diff
@@ -577,26 +494,18 @@ class AppConfig extends Component {
...
-          >
+          <Span size="18" weight="bold" mt="30" ml="38">
             Configure {app.name}
-          </span>
+          </Span>
         )}
-        <div className="flex-column">
-          <div
+        <Flex gap="20px">
+          <SideNavWrapper
```
Comment (miaawong) [inline web/src/features/AppConfig/components/AppConfig.jsx:507]: updated a few divs into styled components

File: web/src/features/AppConfig/styles/index.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import styled from "styled-components";
+import * as colors from "../../../styles/colors";
+
+export const GroupTitle = styled.a`
```
Comment (miaawong) [inline web/src/features/AppConfig/styles/index.js:4]: styled components for AppConfig

File: web/src/styles/common.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  display: flex;
+  flex-direction: ${(props) => props.direction || "row"};
+  flex-wrap: ${(props) => props.wrap || "nowrap"};
+  justify-content: ${(props) => props.justifyContent || "flex-start"};
+  justify-items: ${(props) => props.justifyItems || "flex-start"};
+  align-items: ${(props) => props.align || "flex-start"};
+  gap: ${(props) => props.gap};
+  padding: ${(props) => props.p && `${props.p}px`};
+  padding-top: ${(props) =>
+    (props.pt && `${props.pt}px`) || (props.px && `${props.py}px`)};
```
Comment (miaawong) [inline web/src/styles/common.js:14]: will refactor later.

File: web/src/features/AppConfig/styles/index.js (hunk 6 above)
Comment (rianfowler) [reply web/src/features/AppConfig/styles/index.js:4]: nice

**Review by rianfowler — APPROVED** [review]
Great work!

**Pull request merged**